packages/source-map-utils/index.js

```diff
@@ -17,21 +17,20 @@ var SourceMapUtils = {
     var column = 0;
 
     source.forEach(function (character) {
+      let loc = { line, column };
       if (character === "\n") {
         line += 1;
         column = -1;
 
-        mapping.push({
+        loc = {
           line: line,
           column: 0
-        });
-      } else {
-        mapping.push({
-          line: line,
-          column: column
-        });
+        };
       }
 
+      for (let i = 0; i < Buffer.from(character).length; i++) {
```

Comment:
I'd suggest adding a comment on this loop.  Something like, "because our offsets are given in bytes rather than in characters, we pad out the line/column map as per the UTF-8 length of the character".  I think that will make this clearer for future maintainers.